editoast/src/views/v2/timetable/stdcm.rs

```diff
@@ -209,18 +203,26 @@ async fn stdcm(
         timetable_id,
     )
     .await?;
-    let maximum_run_time = match maximum_run_time_result {
-        MaxRunningTimeResult::MaxRunningTime { value } => value,
-        MaxRunningTimeResult::Error { error } => {
+    let simulation_run_time = match simulation_run_time_result {
+        SimulationTimeResult::SimulationTime { value } => value,
+        SimulationTimeResult::Error { error } => {
             return Ok(Json(STDCMResponse::PreprocessingSimulationError {
                 error: *error,
             }))
         }
     };
+    let earliest_step_tolerance_window = get_earliest_step_tolerance_window(&data);
+    let maximum_departure_delay =
+        get_maximum_departure_delay(&data, simulation_run_time, earliest_step_tolerance_window);
+    let maximum_run_time_without_tolerance = 2 * simulation_run_time + get_total_stop_time(&data);
+    let maximum_run_time = get_maximum_run_time(
+        &data,
+        maximum_run_time_without_tolerance,
+        earliest_step_tolerance_window,
+    );
```

Comment:
There's quite a lot going on between the parsing of the request, the default values and the debug overrides. I'd suggest to wrap all these values in a dedicated structure like:

```rust
struct StdcmTimings {
    maximum_run_time: u64,
    maximum_run_time_without_tolerance: u64,
    ...
}
```

and `let timings = StdcmTimings::from(&data)`. This would:

* gather all these operations in one place, shortening the endpoint code
* make explicit the values that are computed / overridden
* create a place to document the whole process with your schema
* allow us to use `StdcmTimings::has_debug_overrides(&data)` in order to check whether we need to ensure the `StdcmDebug` user role

Of course since this PR is rather urgent this can be left ~~as an exercise for the reader~~ for another PR (which would be another core-related good first issue, maybe for Loup?).

Reply:
It is a good first issue, and I'd rather do it while not hard ressed with time. Issue created: https://github.com/OpenRailAssociation/osrd/issues/8507.